### ISSUE TITLE
Axios sending corrupted binary data from Node.js to client/react when fetching a binary file

### ISSUE THREAD

**Issue by jyotirmoypan** [axios/axios#3921]
#### Describe the bug
Fetching images in Node.js and passing the binary data to client for creating object URL and preview images results binary data to get corrupted

#### To Reproduce
 

1. Fetch a binary file from node.js
2. Forward the binary data to react/client
3. Try to createObjectUrl from the binaryData and open the file in new tab
4. Browser will throw error that the file is corrupted

```js

**Node.js Code**
    axios
        .request(requestOptions)
        .then((response) => {
          global.logger.logMethodExecutionTime("previewDocument", hrstart);
          global.logger.info(
            "document::previewDocument service function ends with success....."
          );
          res.setHeader("Content-Disposition", "inline");
          res.setHeader("Content-Type", response.headers["content-type"]);
          res.status(response.status).send(response.data);
        })
        .catch((error) => {
          global.logger.logMethodExecutionTime("previewDocument", hrstart);
          global.errorHandler.getDefaultErrorResponse({
            err: error,
            message: "document::previewDocument error in api call...",
          });
          errorHandler(error, req, res);
        });
        
**Client side/react code**
   // Using fetch in react and converting response to blob
    previewDocument()
      .then((res) => {
        console.log(res);
        const urlToPreview = URL.createObjectURL(res);
        window.open(urlToPreview, "_blank");
      })
      .catch((err) => {
        console.log("Preview failed", err);
      });        
```

#### Expected behavior
The binary data received by client should be correct and file size same as original file. For now, the file size is higher than original file.

#### Environment
 - Axios Version: 0.21.1
 - Adapter [e.g. XHR/HTTP]
 - Browser : Chrome, Safari
 - Browser Version: Chrome-92.0
 - Node.js Version- 14.14.1
 - OS: Windows 10
 - Additional Library Versions React 16.8

#### Additional context/Screenshots
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by imjustzealous** [comment]
Hello, can you get a screenshot of request header and also the response header?

**Comment by jyotirmoypan** [comment]
@imjustzealous 

**Request headers:** 

Authorization:'Bearer **7cf984-e988-360e-8bbe-d998f43aa***'
responseType:'arrayBuffer'
x-b3-traceid:'2f1ea52c86f25gec'
method:'GET'

**Response Headers**

access-control-allow-headers:'authorization,Access-Control-Allow-Origin,Content-Type,SOAPAction'
access-control-allow-methods:'GET'
access-control-allow-origin:'*'
connection:'close'
content-type:'image/jpeg;charset=UTF-8'
date:'Fri, 06 Aug 2021 20:05:25 GMT'
transfer-encoding:'chunked'
vary:'accept-encoding'
x_correlation_id:'adajhdjahdjahsd'


Please note with the code I sometimes see a small PDF( < 5KB) loading correctly with the object URL. But when I download the content or try to preview larger files the binary data gets corrupted.

**Comment by imjustzealous** [comment]
Hello, I tried to test the axios for fetching image and video , and its working fine
but I think you're missing the `data` property on `URL.createObjectURL()`

**Comment by imjustzealous** [comment]
You can also try to use this code to see if its working, i used this to test for fetching image
```js
axios.get("/some-image-path", {
  responseType: 'blob'
}).then(
  (ok) => {
    console.log('success')
    let theURLObject = URL.createObjectURL(ok.data)
    window.open(theURLObject, "_blank")
  }
)
.catch(
  (error) => {
    console.warn(error)
  }
)
```

**Comment by imjustzealous** [comment]
Hello, sorry for the response earlier, I tried to reproduce the steps by fetching image and sending it to the client, and it works
Feel free to use the code below
```js
// ===============
// Server-side Code
// ===============
if (req.url == "/example") {
    res.writeHead(200, {
      'Content-Type' : 'image/jpeg;charset=UTF-8'
    })
    axios.get("the-image-path", {
      responseType: 'arraybuffer'
    }).then(
      (ok) => {
        res.write(ok.data)
        res.end()
        console.log("sending")
        return;
      }
    )
    .catch(
      (err) => {
        console.log(err)
      }
    )
  }

// ===============
// Client-Side Code
// ===============

axios.get("/example", {
  responseType: 'blob'
})
.then(
  (some) => {
    console.log('success')
    console.log(some)
    let theURLObj = URL.createObjectURL(some.data)
    let theIMGfirst = document.createElement('img')
    theIMGfirst.width=500
    theIMGfirst.height=500
    theIMGfirst.src = theURLObj
    document.getElementById("container").appendChild(theIMGfirst)
  }
)
.catch(
  (error) => {
    console.warn(error)
  }
)
```

**Comment by AmirHmZz** [comment]
It seems `axios` does not support `transfer-encoding:'chunked'` correctly. according to [**MDN Docs**](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding#directives) :

> Data is sent in a series of chunks. The [Content-Length](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Length) header is omitted in this case and at the beginning of each chunk you need to add the length of the current chunk in hexadecimal format, followed by '\r\n' and then the chunk itself, followed by another '\r\n'. The terminating chunk is a regular chunk, with the exception that its length is zero. It is followed by the trailer, which consists of a (possibly empty) sequence of header fields.

But `axios` doesn't process chunks length properly and put them directly into the request's result. Also it won't remove `/r/n` parts from the response.

**Comment by jasonsaayman** [comment]
Hi 👋 

Please could you retry with the latest version and open a new issue should this error still be relevant?

Thanks

**Comment by MaximeConan** [comment]
> ```js
> {
>       responseType: 'arraybuffer'
>     }
> ```

Was stuck on a it since a day, your answer works like a charm for me.

**Comment by teddybee** [comment]
The problem still exist with the newest axios. I got binary blob instead of json list.

**Comment by jasonsaayman** [comment]
Please try 1.2.1

**Comment by adedade** [comment]
> Please try 1.2.1

Worked for me, thanks!

**Comment by qinst64** [comment]
same problem at v1.2, solved when updated to latest version (now v1.4.0)
